parpack --- add support for trilinos mpi blockvector

asartori86 opened the pull request.

davydden commented on include/deal.II/lac/parpack_solver.h:
```diff
@@ -234,6 +234,12 @@ class PArpackSolver : public Subscriptor
   /**
+   * Initialize internal variables when working with BlockVectors
+   */
+  void reinit(const dealii::IndexSet &locally_owned_dofs,
```
don't need `dealii::` namespace here.

davydden commented on include/deal.II/lac/parpack_solver.h:
```diff
@@ -234,6 +234,12 @@ class PArpackSolver : public Subscriptor
   void reinit(const dealii::IndexSet &locally_owned_dofs );
 
   /**
+   * Initialize internal variables when working with BlockVectors
```
please add some description on `@p locally_owned_dofs` and `@p block`

davydden commented on include/deal.II/lac/parpack_solver.h:
```diff
@@ -516,6 +522,54 @@ void PArpackSolver<VectorType>::reinit(const dealii::IndexSet &locally_owned_dof
+  select.resize (ncv, 0);
+
+  // deal.II vectors:
+  src.reinit (block,mpi_communicator);
```
everything but this line is the same for the two functions. I would suggest to refactor the code (maybe add a function in` private:` which is has the common part and then the two functions would call this private one.

davydden commented:
an entry to `changes.h` would be good to have (btw, the support is for block vectors in general, not only for trilinos).

asartori86 commented:
@davydden, thanks for your feedbacks. I've addressed your comments, at least I hope :D 

May I ask a question related to Arpack/PArpack? I was computing the eigenvalues of the laplacian in 2d on a square with side of lenght \pi. In order to get the **smallest** eigenvalues I had to choose **largest_magnitude**. Is this expected?

davydden commented:
thanks, look good to me :+1:

davydden commented:
as for your question, since we do shift-and-invert spectral transformation, then, say for zero shift and positive eignevalues, the largest eigenvalue of the transformed problem will correspond to the smallest eigenvalue of the original problem

asartori86 commented:
thanks @davydden for your explanation!
